From 0d699c47cfea6e3884965c87b9d37374577b8695 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Thu, 31 Oct 2019 15:31:27 +0530
Subject: [PATCH 398/767] octeontx2-pf: Use 2048 byte receive buffers

commit 556600410db1e2490ebf8d46b8f4419b580d8e21 from
git@git.assembla.com:cavium/WindRiver.linux.git

Use 2048 byte buffers for better last level cache
utilization or data distribution across regions.

Change-Id: I726bc8919afb8f52bca7da1ff0facbd3cfc378df
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18078
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h
index 32ddd552e85b..b898693ac08d 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_txrx.h
@@ -21,10 +21,15 @@
 
 #define DMA_BUFFER_LEN	1536 /* In multiples of 128bytes */
 #define OTX2_DATA_ALIGN(X)	ALIGN(X, OTX2_ALIGN)
-#define RCV_FRAG_LEN		\
+#define RCV_FRAG_LEN1		\
 	((OTX2_DATA_ALIGN(DMA_BUFFER_LEN + NET_SKB_PAD)) + \
 	(OTX2_DATA_ALIGN(sizeof(struct skb_shared_info))))
 
+/* Prefer 2048 byte buffers for better last level cache
+ * utilization or data distribution across regions.
+ */
+#define RCV_FRAG_LEN		((RCV_FRAG_LEN1 < 2048) ? 2048 : RCV_FRAG_LEN1)
+
 #define OTX2_HEAD_ROOM		OTX2_ALIGN
 
 #define	OTX2_ETH_HLEN		(VLAN_ETH_HLEN + VLAN_HLEN)
-- 
2.31.1

